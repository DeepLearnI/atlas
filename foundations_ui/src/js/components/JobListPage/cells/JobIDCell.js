import { CopyToClipboard } from 'react-copy-to-clipboard';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';

import 'react-toastify/dist/ReactToastify.css';

class JobIDCell extends Component {
  constructor(props) {
    super(props);

    this.state = {
      jobID: this.props.jobID,
      isError: this.props.isError,
      rowNumber: this.props.rowNumber,
    };
  }

  notifiedCopy(e) {
    e.stopPropagation();
    toast.info('Job ID successfully copied', {
      autoClose: 1500,
      draggable: false,
    });
  }

  render() {
    const {
      jobID, isError, rowNumber,
    } = this.state;

    const aClass = isError
      ? `job-id-cell error row-${rowNumber}`
      : `job-id-cell row-${rowNumber}`;

    // const href = '/'.concat(jobID);
    return (
      <span key={jobID}>
        <span className={aClass}>{jobID}</span>
        <CopyToClipboard text={jobID}>
          <span
            onClick={this.notifiedCopy}
            className="i--icon-copy"
            role="presentation"
          />
        </CopyToClipboard>
      </span>
    );
  }
}

JobIDCell.propTypes = {
  jobID: PropTypes.string,
  isError: PropTypes.bool,
  rowNumber: PropTypes.number,
};

JobIDCell.defaultProps = {
  jobID: '',
  isError: false,
  rowNumber: 0,
};

export default JobIDCell;
