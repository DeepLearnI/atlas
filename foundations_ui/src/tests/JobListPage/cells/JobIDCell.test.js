import React from 'react';
import ReactDOM from 'react-dom';
import JobIDCell from '../../../js/components/JobListPage/cells/JobIDCell';
import { shallow } from 'enzyme';
import configureTests from '../../setupTests';

configureTests();

it('Shallow Renders Job ID Cell', () => {
  const wrapper = shallow(<JobIDCell/>);
  expect(wrapper).toMatchSnapshot();
});