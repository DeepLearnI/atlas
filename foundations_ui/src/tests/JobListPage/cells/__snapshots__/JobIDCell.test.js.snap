// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Job ID Cell 1`] = `
<span
  key=""
>
  <span
    className="job-id-cell row-0"
  />
  <CopyToClipboard
    text=""
  >
    <span
      className="i--icon-copy"
      onClick={[Function]}
      role="presentation"
    />
  </CopyToClipboard>
</span>
`;
